connectors/citrus-openapi/src/main/java/org/citrusframework/openapi/validation/OpenApiRequestValidator.java

```diff
@@ -0,0 +1,97 @@
+ */
+
+package org.citrusframework.openapi.validation;
+
+import static org.citrusframework.openapi.OpenApiSettings.isRequestValidationEnabledlobally;
+
+import com.atlassian.oai.validator.OpenApiInteractionValidator;
+import com.atlassian.oai.validator.model.Request;
+import com.atlassian.oai.validator.model.SimpleRequest;
+import com.atlassian.oai.validator.report.ValidationReport;
+import java.util.ArrayList;
+import java.util.Collection;
+import org.citrusframework.exceptions.ValidationException;
+import org.citrusframework.http.message.HttpMessage;
+import org.citrusframework.http.message.HttpMessageHeaders;
+import org.citrusframework.openapi.model.OperationPathAdapter;
+
+/**
+ * Specific validator that uses atlassian and is responsible for validating HTTP requests
+ * against an OpenAPI specification using the provided {@code OpenApiInteractionValidator}.
+ */
+public class OpenApiRequestValidator extends OpenApiValidator {
+
+    public OpenApiRequestValidator(OpenApiInteractionValidator openApiInteractionValidator) {
+        super(openApiInteractionValidator, isRequestValidationEnabledlobally());
+    }
+
+    @Override
+    protected String getType() {
+        return "request";
+    }
+
+    public void validateRequest(OperationPathAdapter operationPathAdapter,
+        HttpMessage requestMessage) {
+
+        if (enabled && openApiInteractionValidator != null) {
+            ValidationReport validationReport = openApiInteractionValidator.validateRequest(
+                createRequestFromMessage(operationPathAdapter, requestMessage));
+            if (validationReport.hasErrors()) {
+                throw new ValidationException(
+                    constructErrorMessage(operationPathAdapter, validationReport));
+            }
+        }
+    }
+
+    Request createRequestFromMessage(OperationPathAdapter operationPathAdapter,
+        HttpMessage httpMessage) {
+        var payload = httpMessage.getPayload();
+
+        String contextPath = operationPathAdapter.contextPath();
+        String requestUri = (String) httpMessage.getHeader(HttpMessageHeaders.HTTP_REQUEST_URI);
+        if (contextPath != null && requestUri.startsWith(contextPath)) {
+            requestUri = requestUri.substring(contextPath.length());
+        }
+
+        SimpleRequest.Builder requestBuilder = new SimpleRequest.Builder(
+            httpMessage.getRequestMethod().asHttpMethod().name(), requestUri
+        );
+
+        if (payload != null) {
+            requestBuilder = requestBuilder.withBody(payload.toString());
+        }
+
+        SimpleRequest.Builder finalRequestBuilder = requestBuilder;
+        finalRequestBuilder.withAccept(httpMessage.getAccept());
+
+        httpMessage.getQueryParams()
+            .forEach((key, value) -> finalRequestBuilder.withQueryParam(key, new ArrayList<>(
+                value)));
+
+        httpMessage.getHeaders().forEach((key, value) -> {
+            if (value instanceof Collection<?>) {
```

Comment:
I've changed this to:

```suggestion
            if (value instanceof Collection<?> collection) {
```